e2e-tests/expose-sharded/run

```diff
@@ -0,0 +1,232 @@
+
+	desc "Compare mongo config"
+	if [[ ${enable_expose} == "true" ]]; then
+		cfg_0_endpoint=$(get_service_ip $cluster-cfg-0 'cfg')
+		rs0_0_endpoint=$(get_service_ip $cluster-rs0-0)
+	else
+		cfg_0_endpoint="$cluster-cfg-0.$cluster-cfg.$namespace.svc.cluster.local"
+		rs0_0_endpoint="$cluster-rs0-0.$cluster-rs0.$namespace.svc.cluster.local"
+	fi
+
+	rs0_0_endpoint_actual=$(run_mongo 'var host;var x=0;rs.conf().members.forEach(function(d){ if(d.tags.podName=="some-name-rs0-0"){ host=rs.conf().members[x].host;print(host)};x=x+1; })' "clusterAdmin:clusterAdmin123456@${cluster}-rs0.${namespace}" | egrep -v 'I NETWORK|W NETWORK|Error saving history file|Percona Server for MongoDB|connecting to:|Unable to reach primary for set|Implicit session:|versions do not match|bye')
+	cfg_0_endpoint_actual=$(run_mongo 'var host;var x=0;rs.conf().members.forEach(function(d){ if(d.tags.podName=="some-name-cfg-0"){ host=rs.conf().members[x].host;print(host)};x=x+1; })' "clusterAdmin:clusterAdmin123456@${cluster}-cfg.${namespace}" | egrep -v 'I NETWORK|W NETWORK|Error saving history file|Percona Server for MongoDB|connecting to:|Unable to reach primary for set|Implicit session:|versions do not match|bye')
+	if [[ $rs0_0_endpoint_actual != "$rs0_0_endpoint:27017" || $cfg_0_endpoint_actual != "$cfg_0_endpoint:27017" ]]; then
+		desc "Actual values rs $rs0_0_endpoint_actual and cfg $cfg_0_endpoint_actual do not match expected rs $rs0_0_endpoint:27017 and cfg $cfg_0_endpoint:27017"
+		exit 1
+	fi
+
+}
+
+function expose_cluster() {
+	expose_type=$1
+	expose_status=${2:-true}
+
+	kubectl_bin patch psmdb ${cluster} --type=json --patch '[
+          {
+              "op": "replace",
+              "path": "/spec/replsets/0/expose",
+        "value": {
+          "enabled": '$expose_status',
+          "exposeType" : "'"${expose_type}"'"
+        }
+          },
+        {
+              "op": "replace",
+              "path": "/spec/sharding/mongos/expose",
+        "value": {
+          "enabled": '${expose_status}',
+          "exposeType" : "'"${expose_type}"'"
+        }
+          },
+          {
+              "op": "replace",
+              "path": "/spec/sharding/configsvrReplSet/expose",
+        "value": {
+          "enabled": '${expose_status}',
+          "exposeType" : "'"${expose_type}"'"
+        }
+          }]'
+}
+function main() {
+
+	create_infra "$namespace"
+	desc 'create first PSMDB cluster'
+	cluster="some-name"
+
+	kubectl_bin apply \
+		-f "$conf_dir/secrets.yml" \
+		-f "$conf_dir/client.yml"
+
+	apply_s3_storage_secrets
+	if version_gt "1.19" && [ $EKS -ne 1 ]; then
+		cat "$conf_dir/container-rc.yaml" | $sed 's/docker/runc/g' | kubectl_bin apply -f -
+	else
+		kubectl_bin apply -f "$conf_dir/container-rc.yaml"
+	fi
+
+	apply_cluster "$test_dir/conf/$cluster-rs0.yml"
+	desc 'check if all 3 Pods started'
+	wait_for_running $cluster-rs0 3
+	wait_for_running $cluster-cfg 3 "false"
+	wait_for_running $cluster-mongos 3
+
+	sleep 10
```

Comment:
please add comment why we wait and maybe instead of `sleep` we want to use wait_cluster_consistency

Reply:
Ok!